service/controller/resource/azuremachinepoolconditions/conditionvmssready.go

```diff
@@ -0,0 +1,157 @@
+package azuremachinepoolconditions
+
+import (
+	"context"
+	"fmt"
+
+	azureconditions "github.com/giantswarm/apiextensions/v3/pkg/conditions/azure"
+	"github.com/giantswarm/microerror"
+	capzexp "sigs.k8s.io/cluster-api-provider-azure/exp/api/v1alpha3"
+	capi "sigs.k8s.io/cluster-api/api/v1alpha3"
+	capiconditions "sigs.k8s.io/cluster-api/util/conditions"
+
+	"github.com/giantswarm/azure-operator/v5/service/controller/key"
+)
+
+const (
+	vmssDeploymentPrefix               = "nodepool-"
+	VMSSNotFoundReason                 = "VMSSNotFound"
+	VMSSIDNotSetReason                 = "VMSSIDNotSet"
+	VMSSProvisioningStatePrefix        = "VMSSProvisioningState"
+	VMSSProvisioningStateUnknownReason = "VMSSProvisioningStateUnknown"
+	VmssProvisioningStateSucceeded     = "Succeeded"
+	VmssProvisioningStateFailed        = "Failed"
+)
+
+func (r *Resource) ensureVmssReadyCondition(ctx context.Context, azureMachinePool *capzexp.AzureMachinePool) error {
+	r.logDebug(ctx, "ensuring condition %s", azureconditions.VMSSReadyCondition)
+
+	// Get Azure deployments client
```

Comment:
uber nitpick: Not a fan of these comments, let me tell ya :smile_cat:

Reply:
Agreed :smile: It happens that the code and the comment were bigger, but then I removed some code, and also removed some comments or parts of the comment, and then I end up with `// this is a comment` stuff :smile: